### PR TITLE
Dockerize Webapp

### PR DESCRIPTION
**Pull request by beekley**
addresses #326 (semi-related)

# Motivation and context
To run end-to-end tests, we'll have to have a real instance of the app running. Docker will help simplify this process by handling the setup/build/running of the app. It will also make the development environments more consistent since everyone will run the same environment in a container.

# What I did
- Update the `Dockerfile` for the node webapp, which builds the dependencies in the container while keeping the code local to the host. This allows `nodemon` to continue to work.
- Reorganize the config files to support an additional dev environment (`dev-docker`).
- Update the readme to use docker. This could like be simplified further by moving the frontend/mongo setup steps to the `Dockerfile`.

I'd like a reviewer to test running docker on their machine.

### CODE REVIEW

**Comment by beekley** [comment]
Looks redundant w/ https://github.com/helloGov/webapp/pull/334

**Comment by beekley** [comment]
Built on #334

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# Copy container-built node_modules.
+# From: https://stackoverflow.com/a/52092711
+
+cp -r /usr/src/cache/node_modules/. /usr/src/app/node_modules/
+# exec npm run build
+exec npm run start-docker
```
Comment (lllleonnnn) [inline entrypoint.sh:8]: does using the entrypoint allow you to watch/rebuild during local development?

File: index.js [hunk 2]
```diff
@@ -23,33 +23,18 @@ var routes = require('./app/routes');
...
-const currentEnv = app.get('env');
-const localMongoUri = `mongodb://localhost:27017/hellogov`;
-const mongoUri = `mongodb://${config.dbUser}:${config.dbPassword}@${config.db}-shard-00-00-5sypa.mongodb.net:27017,${config.db}-shard-00-01-5sypa.mongodb.net:27017,${config.db}-shard-00-02-5sypa.mongodb.net:27017/${config.db}-${config.dbStage}?ssl=true&replicaSet=helloGov-shard-0&authSource=admin&retryWrites=true`;
-
-if (currentEnv === 'development') {
-    mongoose.connect(localMongoUri);
-} else {
-    mongoose.connect(mongoUri, { useNewUrlParser: true });
-}
+mongoose.connect(config.mongoUri, config.mongoOptions);
```
Comment (lllleonnnn) [inline index.js:26]: 😂

File: index.js [hunk 3]
```diff
@@ -23,33 +23,18 @@ var routes = require('./app/routes');
...
-        touchAfter: 0
-    });
-} else {
-    var sessionStore = new MongoStore({
-        url: localMongoUri,
-        touchAfter: 0
-    });
-}
+var sessionStore = new MongoStore({
+    url: config.mongoUri,
```
Comment (lllleonnnn) [inline index.js:35]: thank you

File: package.json [hunk 4]
```diff
@@ -8,6 +8,7 @@
     "seed": "mongoimport -c users --uri mongodb://localhost:27017/hellogov --drop --file conf/seeds.json",
     "start": "NODE_ENV=production node index.js",
     "start-dev": "NODE_ENV=development nodemon index.js",
+    "start-docker": "NODE_ENV=dev-docker nodemon index.js",
```
Comment (lllleonnnn) [inline package.json:11]: ok here is the watch

**Pull request closed without merging**